lib/bacnet-asn1.js

```diff
@@ -779,7 +784,14 @@ var decode_read_access_result = module.exports.decode_read_access_result = funct
         var localResult = bacapp_decode_application_data(buffer, offset + len, apduLen + offset - 1, value.objectIdentifier.type, new_entry.propertyIdentifier);
         if (!localResult) return;
         len += localResult.len;
-        localValues.push({value: localResult.value, type: localResult.type});
+        var resObj = {
+          value: localResult.value,
+          type: localResult.type
+        };
+        if (localResult.encoding !== undefined) {
```

Comment:
Thought about it too, but then I was thinking that the client doesn't really care about the encoding since the string is already in UTF-8 when sent via JSON and it's more an internal thing. And wrapping value again in another object is again a bit unnecessary or not? I agree, the code here wouls be much cleaner and we could remove all the ifs.